.github/workflows/tox.yml

```diff
@@ -1,22 +1,20 @@
 name: tox
-on: [push, pull_request]
+on:
+  push:
+    branches: [master]
+  pull_request:
 jobs:
   tox:
     strategy:
       fail-fast: false
-      max-parallel: 4
+      max-parallel: 5
       matrix:
-        python: [3.7, 3.8, 3.9]
+        python: ["3.8", "3.9"]  # TODO: Add these... , "3.10", "3.11", "3.12"]
```

Comment:
You can add them, but upgrade pytest in requirements.txt to latest version (7.4.3).